decidim-core/app/presenters/decidim/admin_log/user_presenter.rb

```diff
@@ -49,19 +80,57 @@ def current_justification
         action_log.extra.dig("extra", "current_justification") || Hash.new("")
       end
 
-      # Overwrite the changeset for officialization and block actions.
+      # Overwrite the changeset for officialization,block and more actions.
       def changeset
-        original = { badge: [previous_user_badge, user_badge] }
-        fields = { badge: :i18n }
-        if action.to_s == "block"
-          original = { justification: [previous_justification, current_justification] }
-          fields = { justification: :string }
-        end
-        Decidim::Log::DiffChangesetCalculator.new(original, fields, i18n_labels_scope).changeset
+        config = changeset_config[action.to_s]
+        return {} unless config
+
+        original = config[:original].call
+        return {} if original.values.flatten.all?(&:empty?)
+
+        Decidim::Log::DiffChangesetCalculator.new(original, config[:fields], i18n_labels_scope).changeset
+      end
+
+      def changeset_config
+        {
+          "block" => {
+            original: -> { { justification: [previous_justification, current_justification] } },
+            fields: { justification: :string }
+          },
+          "bulk_block" => {
+            original: -> { { justification: [blocked_users.join(", "), current_justification] } },
+            fields: { justification: :string }
+          },
+          "bulk_unblock" => {
+            original: -> { { unblocked_users: ["", unblocked_users.join(", ")] } },
+            fields: { unblocked_users: :string }
+          },
+          "bulk_ignore" => {
+            original: -> { { unreported_users: ["", unreported_users.join(", ")] } },
+            fields: { unreported_users: :string }
+          },
+          "bulk_hide" => {
+            original: -> { { hidden_content: ["", hidden_content.join(", ")] } },
+            fields: { hidden_content: :string }
+          },
+          "bulk_unreport" => {
+            original: -> { { unreported_content: ["", unreported_content.join(", ")] } },
+            fields: { unreported_content: :string }
+          },
+          "bulk_unhide" => {
+            original: -> { { unhidden_content: ["", unhidden_content.join(", ")] } },
+            fields: { unhidden_content: :string }
+          }
```

Comment:
"Hide" and "unhide" are not actions that you can perform on a user. Let's remove this.
We will need to add the information in different presenters (the moderated content)

Reply:
Done